.github/workflows/test-docker-build.yml

```diff
@@ -0,0 +1,40 @@
+name: Test building a docker image
```

Comment:
Let's just call this `Test docker build` (aligns with the file name and I think it's cleaner; this will be rendered in UX).

Also we might consider having this just be part of the `test.yml` file which would mean on CI it could show up as `Test (docker build)`

Then the name of the could just be `docker-build` since the job is already a subset of test ("discard" feels redundant / implied in that context, which is part of why I think that context might be worth considering!)